src/modules/navigator/geofence.cpp

```diff
@@ -133,38 +104,38 @@ bool Geofence::inside(double lat, double lon, float altitude)
 {
 	bool inside_fence = true;
 
-	float max_horizontal_distance = _param_max_hor_distance.get();
-	float max_vertical_distance = _param_max_ver_distance.get();
-
-	if (max_horizontal_distance > 1.0f || max_vertical_distance > 1.0f) {
-		if (_home_pos_set) {
-			float dist_xy = -1.0f;
-			float dist_z = -1.0f;
-			get_distance_to_point_global_wgs84(lat, lon, altitude,
-							   _home_pos.lat, _home_pos.lon, _home_pos.alt,
-							   &dist_xy, &dist_z);
-
-			if (max_vertical_distance > 1.0f && (dist_z > max_vertical_distance)) {
-				if (hrt_elapsed_time(&_last_vertical_range_warning) > GEOFENCE_RANGE_WARNING_LIMIT) {
-					mavlink_log_critical(_navigator->get_mavlink_log_pub(),
-							     "Maximum altitude above home exceeded by %.1f m",
-							     (double)(dist_z - max_vertical_distance));
-					_last_vertical_range_warning = hrt_absolute_time();
-				}
+	if (isHomeRequired() && _navigator->home_position_valid()) {
+
+		const float max_horizontal_distance = _param_max_hor_distance.get();
+		const float max_vertical_distance = _param_max_ver_distance.get();
+
+		const double home_lat = _navigator->get_home_position()->lat;
+		const double home_lon = _navigator->get_home_position()->lon;
+		const double home_alt = _navigator->get_home_position()->alt;
 
-				inside_fence = false;
+		float dist_xy = -1.0f;
+		float dist_z = -1.0f;
+
+		get_distance_to_point_global_wgs84(lat, lon, altitude, home_lat, home_lon, home_alt, &dist_xy, &dist_z);
+
+		if (max_vertical_distance > 1.0f && (dist_z > max_vertical_distance)) {
```

Comment:
Yes it could. I think these params were originally integers. I'll make the change